resources/js/download.js

```diff
@@ -0,0 +1,105 @@
+/*
+ * activate modal with login form
+ * or logout user 
+ */
+function showModal() {
+    document.getElementById("download-modal").style.display = "block";
+    downloadXmlFiles();
+}
+
+function downloadXmlFiles() {
+    const xhr = new XMLHttpRequest();
+    xhr.open("GET", "/data/list", true);
+    xhr.responseType = "json"; // very important!
+    xhr.onload = async function () {
+        if (xhr.status === 200) {
+            const zip = new JSZip();
+            const files = xhr.response; // use .responseText if not a blob
+            const filteredFiles = files.filter(f => 
+                !f.endsWith('.xql') && f !== '__contents__.xml'
+            );
+            const totalFiles = filteredFiles.length;
+
+            const progressLabel = document.getElementById("download-progress-label");
+            const currentFileLabel = document.getElementById("current-file-label");
+            const progressBar = document.getElementById("progress-bar");
+
+            for (let i = 0; i < totalFiles; i++) {
+                const file = filteredFiles[i];
+                
+                if (!file.endsWith(".xml"))
+                    continue; // skip non-XML files
+
+                progressLabel.textContent = `Downloading file ${i + 1} of ${totalFiles}`;
+                currentFileLabel.textContent = `Currently downloading: ${file}`;
+
+                let resp = await fetch(`/data/read?filename=${encodeURIComponent(file)}`, {
```

Comment:
I would turn this into a relative link:
```suggestion
                let resp = await fetch(`data/read?filename=${encodeURIComponent(file)}`, {
```